src/tree/JobTreeItem.ts

```diff
@@ -0,0 +1,103 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Octokit } from '@octokit/rest';
+import { ActionsGetJobForWorkflowRunResponseData } from '@octokit/types';
+import * as moment from 'moment';
+import * as prettyMs from 'pretty-ms';
+import { AzExtTreeItem, AzureParentTreeItem, IActionContext, TreeItemIconPath } from "vscode-azureextensionui";
+import { createOctokitClient } from '../commands/github/createOctokitClient';
+import { Conclusion, Status } from '../gitHubTypings';
+import { convertConclusionToVerb, convertStatusToVerb, getRepoFullname } from '../utils/gitHubUtils';
+import { localize } from '../utils/localize';
+import { treeUtils } from "../utils/treeUtils";
+import { ActionTreeItem } from './ActionTreeItem';
+import { IAzureResourceTreeItem } from './IAzureResourceTreeItem';
+import { StepTreeItem } from './StepTreeItem';
+
+export class JobTreeItem extends AzureParentTreeItem implements IAzureResourceTreeItem {
+
+    public static contextValue: string = 'azureStaticJob';
+    public readonly contextValue: string = JobTreeItem.contextValue;
+    public parent: ActionTreeItem;
+    public data: ActionsGetJobForWorkflowRunResponseData;
+
+    constructor(parent: ActionTreeItem, data: ActionsGetJobForWorkflowRunResponseData) {
+        super(parent);
+        this.data = data;
+    }
+
+    public get iconPath(): TreeItemIconPath {
+        return treeUtils.getActionIconPath(this.status, this.conclusion);
+    }
+
+    public get id(): string {
+        return this.data.id.toString();
+    }
+
+    public get name(): string {
+        return this.data.name;
+    }
+
+    public get label(): string {
+        return this.name;
+    }
+
+    public get description(): string {
+        if (this.data.conclusion !== null) {
+            const elapsedMs: number = this.completedDate.getTime() - this.startedDate.getTime();
+            return `${convertConclusionToVerb(this.conclusion)} ${moment(this.completedDate).fromNow()} in ${prettyMs(elapsedMs)}`;
+        } else {
+            return `${convertStatusToVerb(this.status)} ${moment(this.startedDate).fromNow()}`;
+        }
+    }
+
+    private get startedDate(): Date {
+        return new Date(this.data.started_at);
+    }
+
+    private get completedDate(): Date {
+        return new Date(this.data.completed_at);
+    }
+
+    private get status(): Status {
+        if (this.data.status in Status) {
+            return <Status>this.data.conclusion;
+        } else {
+            throw new Error(localize('statusNotRecognized', 'Status not recognized.'));
```

Comment:
Add `this.data.status` to the error, so that we have a chance of fixing the error based on telemetry alone. This is what I usually do:
```
throw new RangeError(localize('invalidStatus', 'Invalid status "{0}".', this.data.status));
```